src/daft-micropartition/src/micropartition.rs

```diff
@@ -424,11 +432,15 @@ impl MicroPartition {
         ) {
             // CASE: ScanTask provides all required metadata.
             // If the scan_task provides metadata (e.g. retrieved from a catalog) we can use it to create an unloaded MicroPartition
-            (Some(metadata), Some(statistics), _, _) => Ok(Self::new_unloaded(
-                scan_task.clone(),
-                metadata.clone(),
-                statistics.clone(),
-            )),
+            (Some(metadata), Some(statistics), _, _)
+                if scan_task.pushdowns.filters.is_none() && scan_task.pushdowns.limit.is_none() =>
+            {
+                Ok(Self::new_unloaded(
+                    scan_task.clone(),
+                    metadata.clone(),
```

Comment:
actually we should be able to support unloaded micropartitions with limit pushdowns
```
new_num_rows = min(metadata.num_rows, pushdown.limit)
```

Reply:
I believe materialize scan task takes limit pushdowns into account

Reply:
Oh yes true, forgot that we had access to the actual num rows